homeassistant/components/econet/water_heater.py

```diff
@@ -123,7 +123,7 @@ def set_temperature(self, **kwargs):
 
     def set_operation_mode(self, operation_mode):
         """Set operation mode."""
-        op_mode_to_set = HA_STATE_TO_ECONET.get(operation_mode)
+        op_mode_to_set = HA_STATE_TO_ECONET.get(operation_mode.lower())
```

Comment:
We should enhance the service schema in the water_heater integration instead and only allow the standard operation modes.

https://github.com/home-assistant/core/blob/e65670fef459016b465ab1c8d2ae10732d14803d/homeassistant/components/water_heater/__init__.py#L95

https://github.com/home-assistant/core/blob/e65670fef459016b465ab1c8d2ae10732d14803d/homeassistant/components/water_heater/__init__.py#L51-L56

This is how we do it in the climate integration:
https://github.com/home-assistant/core/blob/e65670fef459016b465ab1c8d2ae10732d14803d/homeassistant/components/climate/__init__.py#L118

Reply:
So there are two paths here: do just the minimum required to improve the service schema, or upgrade the component/integration to use the pattern in the `climate` component. The latter path would probably be better for the long-term, but would break compatibility for any water heater integrations that aren't shipped with the core.

At the moment the water_heater component and integrations use inconsistent naming for 'operation mode' items, and so the latter path would result in consistency.

Reply:
I suggest we don't change anything for now until this architecture discussion is resolved:
https://github.com/home-assistant/architecture/discussions/518

Reply:
OK, I see that... and I was already heading in the same direction. I'll close this PR as there is a much larger effort required.